src/main/java/htsjdk/samtools/SamFileValidator.java

```diff
@@ -323,6 +325,14 @@ private void validateSamRecordsAndQualityFormat(final Iterable<SAMRecord> samRec
                     sequenceDictionaryEmptyAndNoWarningEmitted = false;
 
                 }
+
+                if ((qualityNotStoredErrorCount < MAX_QUALITY_NOT_STORED_ERRORS) && record.getBaseQualityString().equals("*")){
```

Comment:
Is it possible to write a test that verifies that this max error check is working?

Reply:
Test for this case is added.